fractal_graphing/algos/algos.go

```diff
@@ -0,0 +1,98 @@
+// Copyright © 2016 Pawel Rozlach.
+// License: https://creativecommons.org/licenses/by-nc-sa/4.0/
+
+// Based on work by Alan A. A. Donovan & Brian W. Kernighan
+// which can be found at:
+// https://github.com/adonovan/gopl.io.git
+
+// Mandelbrot emits a PNG image of the Mandelbrot fractal.
+package algos
+
+import "math/cmplx"
+
+//(sur) Is it a problem that I make func type private and at the same time
+// map using it public ?
+type AlgoFunc func(r, i float64) (uint8, uint8)
+
+// (sur) Is there a way to do _getattr_ in a simple way ? Or is the "map"
+// approach idiomatic enough ?
+var Str2funcMapping = map[string]AlgoFunc{
```

Comment:
I'd be careful to make this public. This map is mutable hence anyone can change it. I'd recommend making this private.

But otherwise this is a fine approach! Go is meant to be simple ;)